Source/Core/Core/Scripting/HelperClasses/FunctionMetadata.h

```diff
@@ -0,0 +1,63 @@
+#ifndef FUNCTION_METADATA_API_IMPLS
+#define FUNCTION_METADATA_API_IMPLS
+
+#include <string>
+#include "Core/Scripting/CoreScriptInterface/Enums/ArgTypeEnum.h"
+#include "Core/Scripting/HelperClasses/ArgHolder.h"
+
+#include "Core/Scripting/HelperClasses/ScriptContext.h"
+
+// This file contains the implementation for the FunctionMetadata class, and the implementations for
+// the APIs in FunctionMetadata_APIs
+namespace Scripting
+{
+class FunctionMetadata
+{
+public:
+  FunctionMetadata()
+  {
+    this->function_name = "";
+    this->function_version = "";
+    this->example_function_call = "";
+    function_pointer = nullptr;
+    return_type = (ArgTypeEnum)0;
+    arguments_list = std::vector<ArgTypeEnum>();
```

Comment:
This (and the following ctor) seem really inconsistent in style. I don't think we made any decision either way in our code style; but the prevailing style elsewhere is _without_ the `this->` qualification (unless it is necessary to make the unabmiguous; which shouldn't ever be necessary since we prefer classes to use the `m_` prefix for members)

Reply:
Fixed.